tap_anthem/streams.py

```diff
@@ -1,62 +1,149 @@
+
+        # Applicable where type = 'INDIVIDUAL'
+        th.Property("name", th.ObjectType(
+            th.Property("first", th.StringType),
+            th.Property("middle", th.StringType),
+            th.Property("last", th.StringType)
+        )),
+
+        # Applicable where type = 'GROUP'
+        th.Property("group_name", th.StringType),
+
+        # Applicable where type = 'FACILITY'
+        th.Property("facility_name", th.StringType),
+        th.Property("facility_type", th.ArrayType(th.StringType)),
+
+        th.Property("type", th.StringType),
+        th.Property("accepting", th.StringType),
+        th.Property("gender", th.StringType),
+        th.Property("languages", th.ArrayType(th.StringType)),
+        th.Property("specialty", th.ArrayType(th.StringType)),
+        th.Property("addresses", th.ArrayType(
+            th.ObjectType(
+                th.Property("address", th.StringType),
+                th.Property("city", th.StringType),
+                th.Property("state", th.StringType),
+                th.Property("zip", th.StringType),  # String to preserve leading zeroes
+                th.Property("phone", th.StringType)
+            )
+        )),
+        th.Property("last_updated_on", th.DateTimeType),
+        th.Property("plans", th.ArrayType(
+            th.ObjectType(
+                th.Property("plan_id_type", th.StringType),
+                th.Property("plan_id", th.StringType),
+                th.Property("network_tier", th.StringType),
+                th.Property("years", th.ArrayType(th.IntegerType)),
+            )
+        ))
+
     ).to_dict()
 
 
-class GroupsStream(anthemStream):
+class NhPlansStream(AnthemStream):
     """Define custom stream."""
-    name = "groups"
-    path = "/groups"
-    primary_keys = ["id"]
-    replication_key = "modified"
+    name = "nh_plans"
+    path = '/PLANS_NH.json'
+    primary_keys = ["plan_id"]
+    replication_key = "last_updated_on"
+
+    def parse_response(self, response: requests.Response) -> Iterable[dict]:
+
+        for row in response.json():
+            for formulary in row["formulary"]:
+                for costsharing in formulary["costSharing"]:
+                    costsharing["pharmacy_type"] = costsharing["pharmacyType"]
+                    del costsharing["pharmacyType"]
+
+                    costsharing["copay_amount"] = costsharing["copayAmount"]
+                    del costsharing["copayAmount"]
+
+                    costsharing["copay_opt"] = costsharing["copayOpt"]
+                    del costsharing["copayOpt"]
+
+                    costsharing["coinsurance_rate"] = costsharing["coinsuranceRate"]
+                    del costsharing["coinsuranceRate"]
+
+                    costsharing["coinsurance_opt"] = costsharing["coinsuranceOpt"]
+                    del costsharing["coinsuranceOpt"]
+                formulary["cost_sharing"] = formulary["costSharing"]
+                del formulary["costSharing"]
```

Comment:
The keys in the `formulary` array have inconsistent formatting relative to the rest of the document (namely, `camelCase` versus `snake_case`). 

This block of code converts the formatting to `snake_case` and deletes the original `camelCase` key so there's no ambiguity around which key is the 'correct' one to use.

**Why this matters**

For one reason, consistency. 

More importantly: although this tap makes no assumptions about which [loader](https://hub.meltano.com/loaders/) will be used, Postgres is a common choice for smaller warehouses. `camelCase` column names [are not preferred by Postgres](https://wiki.postgresql.org/wiki/Don%27t_Do_This#Don.27t_use_upper_case_table_or_column_names), so rather than cause a headache for end-users when they query this dataset, I'm renaming the keys. 

**Alternate approaches**

Given there are six keys that need to be renamed, I kept things simple and renamed each manually.

However, if a large portion of the keys in the document needed to be renamed, I'd opt for something more automated, perhaps a regular expression.